src/org/traccar/database/DataManager.java

```diff
@@ -334,11 +346,34 @@ public void removeUser(long userId) throws SQLException {
                 .executeQuery(GroupPermission.class);
     }
 
-    public Collection<Device> getAllDevices() throws SQLException {
+    private Collection<Device> getAllDevices() throws SQLException {
         return QueryBuilder.create(dataSource, getQuery("database.selectDevicesAll"))
                 .executeQuery(Device.class);
     }
 
+    public Collection<Device> getAllDevicesCached() {
```

Comment:
Please explain the reason for this function. Why `getDeviceById` and `getDeviceByUniqueId` is not enough?